src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,97 @@
+package storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+import exceptions.MorganException;
+import parser.TaskParser;
+import tasks.Task;
+import tasks.TaskList;
+
+/**
+ * This is a storage.Storage class, which encapsulates storage of tasks.
+ */
+public class Storage {
+    private static final String DIRECTORY_PATH = "./morgan-files";
+    private static final String FILE_PATH = DIRECTORY_PATH + "/tasks.txt";
+    private static final String FILE_ERROR = "Warning: Unable to locate storage data. "
+            + "A new storage file has been created.";
+    private static final String DECODE_ERROR = "Warning: Some tasks may be missing due "
+            + "to storage file tampering. ";
+    private static final String ENCODE_ERROR = "Warning: Unable to store latest update "
+            + "to task data";
+
+    private final TaskParser parser = new TaskParser();
+    private final File file;
+
+    /**
+     * Constructor for Storage.
+     */
+    public Storage() {
+        File directory = new File(DIRECTORY_PATH);
+        boolean isDirectoryExist = directory.exists();
+        if (!isDirectoryExist) {
+            directory.mkdir();
+        }
+        this.file = new File(FILE_PATH);
+    }
+
+    /**
+     * Loads data from specified file.
+     * @param taskList The existing list of tasks from the file.
+     * @throws MorganException If storage data has been tampered.
+     */
+    public void load(TaskList taskList) throws MorganException {
+        boolean hasInvalidFormatting = false;
+        try {
+            Scanner sc = new Scanner(this.file);
+            taskList.clear();
+            while (sc.hasNextLine()) {
+                String taskString = sc.nextLine();
+                try {
```

Comment:
Perhaps you can consider adding multiple catch blocks below the first try block rather than nesting it as the nested catch block is catching a different exception from the main catch block?